### PR TITLE
[CMD] Fix GetPathCase function

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

This `GetPathCase` function does resolve the path with wildcard pattern, and fix the alphabet cases, by using `kernel32!FindFirstFile` function.
Enable wildcard at the middle of the file path in `cmd.exe` by fixing this function.

JIRA issue: [CORE-6609](https://jira.reactos.org/browse/CORE-6609)
JIRA issue: [CORE-13906](https://jira.reactos.org/browse/CORE-13906)

## Proposed changes

- Fix `GetPathCase` function for supporting wildcard at the middle of the file path.
- Make `GetPathCase` string-safe.
- Optimize `GetPathCase`.

## TODO

- [x] Do tests.

## Comparison

BEFORE:
before
FAILED.

AFTER:
after
Successful.

### CODE REVIEW

File: base/shell/cmd/misc.c [hunk 1]
```diff
@@ -83,45 +83,56 @@ cgetchar (VOID)
...
+    *pchTemp = OutPath[0] = 0;
+
+    for (i = 0; i < cchPath; ++i)
     {
-        if (Path[i] != _T('\\'))
+        if (pchTemp + 1 >= pchTempEnd)
         {
-            _tcsncat(TempPath, &Path[i], 1);
-            if (i != _tcslen(Path) - 1)
+            StringCchCopy(OutPath, MAX_PATH, Path);
```
Comment (HBelusca) [inline base/shell/cmd/misc.c:100]: Why doing a straight copy to OutPath in that case?

**Comment by HBelusca** [comment]
@katahiromz Could you test the following:
in `C:\`, besides the ReactOS directory, create another directory named `Reanot` (just a random name that starts with "Rea" and the next letter is > c) and in that new directory, create a `bin` subdirectory.
Then when you're inside `C:\Documents and settings`, run: `cd ..\rea*\bin`.

What happens in that case?

Then, can you do the same test in Windows (XP/2003/10+), using Windows' cmd.exe ? (I suppose there that you don't have a ReactOS directory, so you would need to create a similar layout). I wonder what happens there.

File: base/shell/cmd/misc.c [hunk 2]
```diff
@@ -83,45 +83,58 @@ cgetchar (VOID)
 /*
  * Takes a path in and returns it with the correct case of the letters
  */
-VOID GetPathCase( TCHAR * Path, TCHAR * OutPath)
+VOID GetPathCase(IN LPCTSTR Path, OUT LPTSTR OutPath)
 {
-    UINT i = 0;
+    UINT i;
+    UINT cchPath = _tcslen(Path);
```
Comment (katahiromz) [inline base/shell/cmd/misc.c:89]:
```suggestion
    SIZE_T cchPath = _tcslen(Path);
```

File: base/shell/cmd/misc.c (hunk 1 above)
Comment (katahiromz) [reply base/shell/cmd/misc.c:102]: This is a failed case. To display the failed wildcard pattern, the original text is better.

**Comment by katahiromz** [comment]
The results.

ReactOS (AFTER):
jikken

Win2k3:
jikken-2

**Comment by katahiromz** [comment]
The person in charge is on a long vacation, but we cannot allow progress to be delayed.

File: base/shell/cmd/misc.c (hunk 1 above)
Comment (katahiromz) [reply base/shell/cmd/misc.c:102]:
```suggestion
            // If it fails, copy the original path for an error message
            StringCchCopy(OutPath, MAX_PATH, Path);
```

**Review by HBelusca — APPROVED** [review]
LGTM; final commit message should mention the JIRA report numbers.

**Pull request merged**